### PR TITLE
MPA-13: Fix upload to testflight action.

### PR DESCRIPTION
**Pull request by madalinaardelean**

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -14,25 +14,27 @@ platform :ios do
...
-                method: "appstore",
+                provisioningProfiles: {
+                    ENV["FL_APP_IDENTIFIER"] => ENV['sigh_' + app_identifier + '_appstore_profile-name']
+                },
                 uploadBitcode: false
             }
         )
 
         upload_to_testflight(
+            api_key: lane_context[SharedValues::APP_STORE_CONNECT_API_KEY],
```
Comment (p4checo) [inline fastlane/Fastfile:37]: isn't this set automatically via env var by fastlane already? 🤔

File: fastlane/Fastfile [hunk 2]
```diff
@@ -55,30 +58,24 @@ def fetch_production_code_signing
...
-    sync_code_signing(
-        type: "development",
-        readonly: true,
-        generate_apple_certs: false
-    )
-    update_code_signing_settings(
-        code_sign_identity: "iPhone Development",
-        profile_name: ENV['sigh_' + app_identifier + '__profile-name'],
-        build_configurations: ["Debug"]
+    update_project_provisioning(
```
Comment (p4checo) [inline fastlane/Fastfile:74]: why is this necessary when we already have the `update_code_signing_settings` action before?

Comment (madalinaardelean) [reply fastlane/Fastfile:67]: I need to keep them both because I get errors if I remove either of them.
Without `update_profile_provisioning` I keep getting this error: `error: "MinderaPeople" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'MinderaPeople' from project 'MinderaPeople’)`
Without `update_code_signing` I get this error: `error: MinderaPeople has conflicting provisioning settings. MinderaPeople is automatically signed, but provisioning profile match AppStore com.mindera.people 1654694194 has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor, or switch to manual signing in the Signing & Capabilities editor. (in target 'MinderaPeople' from project 'MinderaPeople’)`.
I cleaned a bit the `update_code_signing` action, since I only need to enable the manual signing.

File: fastlane/Fastfile (hunk 1 above)
Comment (madalinaardelean) [reply fastlane/Fastfile]: Removed it.

File: fastlane/Fastfile (hunk 2 above)
Comment (p4checo) [reply fastlane/Fastfile:67]: Have you tried doing this?

> Set the provisioning profile value to "Automatic" in the build settings editor

On my current project we only use `update_code_signing` like this:

```ruby
    update_code_signing_settings(
      build_configurations: ["Release"],
      code_sign_identity: "iPhone Distribution",
      profile_name: ENV['sigh_' + app_identifier + '_appstore_profile-name']
    )
```

And have the project configured for the Beta/Release config like this:

Screen Shot 2023-01-18 at 16 17 22

Comment (madalinaardelean) [reply fastlane/Fastfile:67]: But if I configure the project like this it would mean I would need to set the provisioning profile and the team as being public in the repo. Do we want this? 
`update_project_provisioning` does exactly this.

Comment (p4checo) [reply fastlane/Fastfile:67]: Oh, that's right... But I think we would only need to set the Team (ID).

So that means that with the current setup while working on the MP app devs always need to explicitly add Mindera's team if they want to deploy to a device, right? Additionally, they must be aware of not committing that change, which is something that can slip quite easily. 😅 

Perhaps we could live with the Team ID being on the project settings? I don't think there's much anyone can do with it without proper credentials... 🤔 Worth considering?

Anyway, the [`update_code_signing_settings`](https://docs.fastlane.tools/actions/update_code_signing_settings/) action allows configuring everything from a single invocation if we want to keep it this way (we probably don't need all parameters, just showing all that are possible):

```ruby
 # more advanced manual code signing
update_code_signing_settings(
  use_automatic_signing: false,
  path: "demo-project/demo/demo.xcodeproj",
  team_id: "QABC123DEV",
  bundle_identifier: "com.demoapp.QABC123DEV",
  code_sign_identity: "iPhone Distribution",
  sdk: "iphoneos*",
  profile_name: "Demo App Deployment Profile",
  entitlements_file_path: "Demo App/generated/New.entitlements"
)
```

Comment (beloso) [reply fastlane/Fastfile:67]: Exposing the Team ID should not be a problem, apparently in some cases it's actually required to expose it (for universal links for example) [[source](https://stackoverflow.com/questions/44417171/should-i-keep-the-apple-developer-team-id-secret)].

I don't mind leaving the TeamID exposed, but perhaps we could have a more in depth discussion about this.

Comment (p4checo) [reply fastlane/Fastfile:67]: That's a great point, @beloso. It does get publicly exposed in `apple-app-site-association` files so it can't be sensitive 😅 

I believe we should try to keep private things private but also minimize friction while developing on a day to day basis, i.e. we shouldn't force devs to always change signing settings to test on a physical device

File: mindera-people/Sources/Services/MinderaPeopleService/Responses/User.swift [hunk 3]
```diff
@@ -7,7 +7,6 @@ public struct User: Equatable {
     public let email: String
 }
 
-#if DEBUG
```
Comment (asos-tiagoveloso) [inline mindera-people/Sources/Services/MinderaPeopleService/Responses/User.swift:10]: Why is this being removed?

Comment (madalinaardelean) [reply mindera-people/Sources/Services/MinderaPeopleService/Responses/User.swift:10]: Because I noticed it was called inside the `noop` property of `MinderaPeopleServiceClient`, which is being used in all environments, not only `DEBUG`, which causes the archive to fai

**Review by beloso — APPROVED** [review]
